x/concentrated-liquidity/swaps.go

```diff
@@ -652,29 +663,33 @@ func (k Keeper) swapCrossTickLogic(ctx sdk.Context,
 	nextInitializedTick int64, nextTickIter db.Iterator,
 	p types.ConcentratedPoolExtension,
 	spreadRewardAccum *accum.AccumulatorObject, uptimeAccums *[]*accum.AccumulatorObject,
-	tokenInDenom string) (SwapState, error) {
+	tokenInDenom string, updateAccumulators bool) (SwapState, error) {
 	nextInitializedTickInfo, err := ParseTickFromBz(nextTickIter.Value())
 	if err != nil {
 		return swapState, err
 	}
-	if *uptimeAccums == nil {
-		uptimeAccumsRaw, err := k.GetUptimeAccumulators(ctx, p.GetId())
-		if err != nil {
-			return swapState, err
+	if updateAccumulators {
+
+		if *uptimeAccums == nil {
+			uptimeAccumsRaw, err := k.GetUptimeAccumulators(ctx, p.GetId())
+			if err != nil {
+				return swapState, err
+			}
+			uptimeAccums = &uptimeAccumsRaw
 		}
-		uptimeAccums = &uptimeAccumsRaw
-	}
 
-	if err := k.updateGivenPoolUptimeAccumulatorsToNow(ctx, p, *uptimeAccums); err != nil {
-		return swapState, err
-	}
+		if err := k.updateGivenPoolUptimeAccumulatorsToNow(ctx, p, *uptimeAccums); err != nil {
+			return swapState, err
+		}
 
-	// Retrieve the liquidity held in the next closest initialized tick
-	spreadRewardGrowth := sdk.DecCoin{Denom: tokenInDenom, Amount: swapState.globalSpreadRewardGrowthPerUnitLiquidity}
-	liquidityNet, err := k.crossTick(ctx, p.GetId(), nextInitializedTick, &nextInitializedTickInfo, spreadRewardGrowth, spreadRewardAccum.GetValue(), *uptimeAccums)
-	if err != nil {
-		return swapState, err
+		// Retrieve the liquidity held in the next closest initialized tick
+		spreadRewardGrowth := sdk.DecCoin{Denom: tokenInDenom, Amount: swapState.globalSpreadRewardGrowthPerUnitLiquidity}
+		_, err := k.crossTick(ctx, p.GetId(), nextInitializedTick, &nextInitializedTickInfo, spreadRewardGrowth, spreadRewardAccum.GetValue(), *uptimeAccums)
```

Comment:
In a subsequent PR I'll refactor crossTick to not return LiquidityNet, it just reads that from one of the args.